### PR TITLE
WELD-2207 fix TODO's and remove environment.servlet.util.ServiceLoader

### PR DESCRIPTION
**Pull request by tremes**

### CODE REVIEW

File: impl/src/main/java/org/jboss/weld/util/ServiceLoader.java [hunk 1]
```diff
@@ -232,27 +232,18 @@ private void loadService(String serviceClassName, URL file, int lineNumber) {
         return serviceClass;
     }
 
-    private S prepareInstance(Class<? extends S> serviceClass) {
-        try {
-            // TODO Support the SM
-            Constructor<? extends S> constructor = serviceClass.getDeclaredConstructor();
-            constructor.setAccessible(true);
-            return constructor.newInstance();
-        } catch (LinkageError e) {
```
Comment (tremes) [inline impl/src/main/java/org/jboss/weld/util/ServiceLoader.java]: I am not sure whether we can remove this.

File: environments/servlet/core/src/main/java/org/jboss/weld/environment/servlet/WeldServletLifecycle.java [hunk 2]
```diff
@@ -64,7 +65,6 @@
 import org.jboss.weld.environment.servlet.deployment.WebAppBeanArchiveScanner;
 import org.jboss.weld.environment.servlet.logging.WeldServletLogger;
 import org.jboss.weld.environment.servlet.services.ServletResourceInjectionServices;
-import org.jboss.weld.environment.servlet.util.ServiceLoader;
```
Comment (mkouba) [inline environments/servlet/core/src/main/java/org/jboss/weld/environment/servlet/WeldServletLifecycle.java]: Hm, I think you cannot just replace the import - it works with `Iterable<Container>` and not `Iterable<Metadata<Container>>`.

File: impl/src/main/java/org/jboss/weld/util/ServiceLoader.java (hunk 1 above)
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/util/ServiceLoader.java]: Hm, I think we should follow the ServiceLoader from JDK, i.e. catch `Throwable` and throw `ServiceConfigurationError`?

Comment (tremes) [inline impl/src/main/java/org/jboss/weld/util/ServiceLoader.java]: Well maybe. TBH I don't know the difference between `LinkageError` and `InstantiationException` in the original class so it's probably easier to throw `ServiceConfigurationError`.

**Comment by mkouba** [comment]
Merged, thanks!

**Pull request closed without merging**